### ISSUE TITLE
Transformer training time

### ISSUE THREAD

**Issue by sajastu** [OpenNMT/OpenNMT-py#1373]
Hi,

In the Transformer paper "Attention is all you need", the authors have claimed that the model can significantly speed up the training time (probably compared to those trained with RNN). 
However, when I set the OpenNMT model architecture to use the Transformer as encoder and decoder, it takes a relatively long time to terminate the training (~25 hrs). While this amount is 12hrs for RNN architecture. I'm working on summarization if that helps.

This is the command I try to use for training:

`python train.py -save_model models/med -data data/med -word_vec_size 100  -encoder_type transformer -decoder_type transformer -pre_word_vecs_enc "data/embeddings.enc.pt" -pre_word_vecs_dec "data/embeddings.dec.pt" -train_steps 200000 -dropout 0.5 -batch_size 4096 -optim adam -learning_rate 2 -batch_type tokens -accum_count 4 -copy_attn -param_init_glorot -param_init 0 -seed 1234 -transformer_ff 2048 -heads 5 -position_encoding -share_embeddings -adam_beta2 0.998 -label_smoothing 0.1 -decay_method noam  -warmup_steps 8000 -rnn_size 100 -world_size 2 -gpu_ranks 0 1 -layers 4 `

Any ideas why the Transformer takes so so much time to train?

**Comment by vince62s** [comment]
The claim in the paper is just there to say it takes less time to achieve the same results.
Comparing training time does not really make sens when you don't put all the parameters on the table.
Size of your transformer, size of your rnn, training steps, Bleu/rouge achievement, ....

**Comment by sajastu** [comment]
@vince62s Thank you for the clue. I changed the parameters to what I had trained with RNN, and for the Transformer-related parameters, I used those tried by Google in the paper. It turns out Transformer architecture is highly sensitive to hyperparameters.

**Comment by eduamf** [comment]
A dropout of 0.5. It seems to me a very high value.
